jax/_src/lax/lax.py

```diff
@@ -4896,7 +4896,10 @@ def _argminmax_translation_rule(value_comparator, identity,
 def _argminmax_gpu_translation_rule(op, a, *, axes, index_dtype):
   axis, = axes
   idxs = tie_in(a, broadcasted_iota(index_dtype, a.shape, axis))
-  maxval = np.array(dtypes.iinfo(index_dtype).max, dtype=index_dtype)
+  maxval = (dtypes.finfo(index_dtype).max
```

Comment:
Actually, on reflection: why do we support inexact index types at all? This seems like a bug. Perhaps a better change would be to disallow them in the type rules?

Reply:
Yes, I was also wondering if it made more sense to add a check in the type rule. By default, since it is possible to align the behaviour on GPU with what was already possible on CPU/TPU, I decided to go that way – I don't know if anyone depends on it. But if you think it is wiser to instead restrict it beforehand, I am also fine with that :)

Reply:
I'm comfortable with disallowing it. I doubt anyone depends on it (it seems like an unintentional "feature") and it is untested.